Stage Type.equal

samoht opened the pull request:
/cc @CraigFe

CraigFe commented on src/irmin-pack/IO.ml:
```diff
@@ -106,7 +106,7 @@ module Unix : S = struct
     readonly : bool;
     version : version;
-    buf : Buffer.t;
+    mutable buf : string list;
```
Looks like an unrelated change here. I got the impression that this ended up being bad for performance?

samoht replied:
hum yes indeed I though I had reverted this but that doesn't seem to be the case. Thanks for noticing!

samoht commented:
I have removed the unrelated changes and added a CHANGELOG. Will probably consolidate this file later on anyway, as the generics have a new signature (which is a breaking change).
